### PR TITLE
Add shadow layers to decouple from light layers

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR
Add shadow layers to decouple from light layers
image
image

---
### Testing status
**Katana Tests**: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=HDRP/feature/shadow_layers&automation-tools_branch=master&unity_branch=trunk

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

Updated light layer tests wuth shadow layers:
2502_LightLayers Bitmask
2501_LightLayers

---
### Overall Product Risks
**Technical Risk**: Low

**Halo Effect**: Low

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.Skin.cs [hunk 1]
```diff
@@ -119,6 +119,10 @@ sealed class Styles
             public readonly GUIContent kernelSize = new GUIContent("Kernel size");
             public readonly GUIContent maxDepthBias = new GUIContent("Max Depth Bias");
 
+            // Layers
+            public readonly GUIContent decoupleLightAndShadowLayersText = new GUIContent("Link Light Layer", "Allow you to use two masks for shadows and light layers");
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.Skin.cs]: @JordanL8 need your help for a better tools tips :) 

We would like to say that this option allow to have two separate light layers (i.e a light layers for lighting and a shadow layers for shadow), this allow to ahve object that can cast shadow but don't receive lighting form a particular light.

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs [hunk 2]
```diff
@@ -586,12 +593,6 @@ static void DrawShadowMapContent(SerializedHDLight serialized, Editor owner)
                 if(LightShape.Rectangle == serialized.editorLightShape)
                 {
                     EditorGUILayout.PropertyField(serialized.serializedLightData.useRayTracedShadows, s_Styles.useRayTracedShadows);
-                    if(serialized.serializedLightData.useRayTracedShadows.boolValue)
-                    {
-                        EditorGUILayout.PropertyField(serialized.serializedLightData.numRayTracingSamples, s_Styles.numRayTracingSamples);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs]: Is this expected ? (remove of this line)

File: com.unity.render-pipelines.high-definition/Editor/Lighting/SerializedHDLight.cs [hunk 3]
```diff
@@ -157,7 +155,11 @@ public SerializedHDLight(HDAdditionalLightData[] lightDatas, AdditionalShadowDat
                     showFeatures = o.Find(x => x.featuresFoldout),
                     showAdditionalSettings = o.Find(x => x.showAdditionalSettings),
                     useVolumetric = o.Find(x => x.useVolumetric),
-                    renderingLayerMask = settings.renderingLayerMask
+                    renderingLayerMask = settings.renderingLayerMask,
+
+                    // Layers
+                    linkLightLayers = o.Find(x => x.decoupleShadowLayers),
+                    lightLayerMask = o.Find(x => x.lightLayerMask)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/SerializedHDLight.cs]: typo: lightLayersMask

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs [hunk 4]
```diff
@@ -207,10 +203,15 @@ public float volumetricDimmer
         [Obsolete("Use Light.renderingLayerMask instead")]
         public LightLayerEnum lightLayers = LightLayerEnum.LightLayerDefault;
 
+        // TODO: another enum name ?
+        // Now the renderingLayerMask is used for shadow layers and not light layers
+        public LightLayerEnum lightLayerMask = LightLayerEnum.LightLayerDefault;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs]: lightlayersMask

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs (hunk 2 above)
Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs]: no, I'll re-add the lines

File: com.unity.render-pipelines.high-definition/Editor/Lighting/SerializedHDLight.cs (hunk 3 above)
Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/SerializedHDLight.cs]: done

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs (hunk 2 above)
Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs]: done

**Comment by iM0ve** [comment]
**Sumary:** Found an issue regarding Undo/Link Light Layer function, fixed a minor scene related issue, updated the 2501 scene with additional objects

**Undo/Link Light Layer issue:**
q8KWxPaIpW
When the Link Light Layer checkbox is set, the undo or unchecking does not restore previous Light Layer settings

**Minor scene related issue:**
Capture
The shadow layer light had Link Light Layer checked, so as I clicked on the light it updated(see above) and did not match the screenshot provided by alelievr. I separated light and shadows to make it match the image.

**Updated the 2501 scene:**
image
Added spheres that only receive light(left sphere) and only cast shadows(right sphere). Test was successful, all sources worked as intended.

**Pull request merged**